### PR TITLE
Room header UI updates

### PR DESCRIPTION
**Pull request by germain-gg**
Fixes https://github.com/vector-im/element-web/issues/25892

This PR does three things, can be reviewed commit by commit

- Add public room indicator
- Format room members count better
- Add search to room summary card

---

It also fixes some performance regressions with the `useRoomMembers` hook. 
With the current implementation it would create a new function, with leading: true, rendering the whole throttling useless- Fix performance issues with useRoomMembers

Screenshot 2023-08-31 at 20 31 07
Screenshot 2023-08-31 at 20 31 31

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Room header UI updates ([\#11507](https://github.com/matrix-org/matrix-react-sdk/pull/11507)). Fixes vector-im/element-web#25892.

### CODE REVIEW

**Comment by andybalaam** [comment]
@germain-gg I just tried this out in the netlify deployment and the threads button and notifications bell don't work for me. They both take me to the room info panel instead of the threads panel or notifications panel respectively.

I'll look at the code in the meantime, but I'm assuming that is not intended?

File: src/i18n/strings/en_EN.json [hunk 1]
```diff
@@ -868,7 +868,7 @@
         "hidebold": "Hide notification dot (only display counters badges)",
         "intentional_mentions": "Enable intentional mentions",
         "ask_to_join": "Enable ask to join",
-        "new_room_decoration_ui": "Under active development, new room header & details interface"
+        "new_room_decoration_ui": "New room header & details interface"
```
Comment (andybalaam) [inline src/i18n/strings/en_EN.json:871]: Isn't this now unused?

**Review by andybalaam — APPROVED** [review]
One question, looks good. The problem with the threads and notifications buttons are not related, so can be fixed outside this PR.

File: src/i18n/strings/en_EN.json (hunk 1 above)
Comment (germain-gg) [reply src/i18n/strings/en_EN.json:871]: Still used here, https://github.com/matrix-org/matrix-react-sdk/pull/11507/files#diff-56fe49f3c6514890fd8404b8bfc9e1351f29839e8dfd98931872e5acab84e2f7L550

**Pull request merged**